File: port_java/com/boare/vsq/UstEvent.java
/*
 * UstEvent.java
 * Copyright (c) 2009 kbinani
 *
 * This file is part of com.boare.vsq.
 *
 * com.boare.vsq is free software; you can redistribute it and/or
 * modify it under the terms of the BSD License.
 *
 * com.boare.vsq is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
 */
package com.boare.vsq;

import java.text.*;
import java.io.*;
import com.boare.corlib.*;

public class UstEvent implements Cloneable {
    public String tag;
    public int length = 0;
    public String lyric = "";
    public int note = -1;
    public int intensity = -1;
    public int pbType = -1;
    public float[] pitches = null;
    public float tempo = -1;
    public UstVibrato vibrato = null;
    public UstPortamento portamento = null;
    public int preUtterance = 0;
    public int voiceOverlap = 0;
    public UstEnvelope envelope = null;
    public String flags = "";
    public int moduration = 100;

    public UstEvent(){
    }

    public static boolean isXmlIgnored( String name ){
        if( name.equals( "tag" ) ){
            return true;
        }
        return false;
    }

    public static String getXmlElementName( String name ){
        if( name.equals( "length" ) ){
            return "Length";
        }else if( name.equals( "lyric" ) ){
            return "Lyric";
        }else if( name.equals( "note" ) ){
            return "Note";
        }else if( name.equals( "intensity" ) ){
            return "Intensity";
        }else if( name.equals( "pbType" ) ){
            return "PBType";
        }else if( name.equals( "pitches" ) ){
            return "Pitches";
        }else if( name.equals( "tempo" ) ){
            return "Tempo";
        }else if( name.equals( "vibrato" ) ){
            return "Vibrato";
        }else if( name.equals( "portamento" ) ){
            return "Portamento";
        }else if( name.equals( "preUtterance" ) ){
            return "PreUtterance";
        }else if( name.equals( "voiceOverlap" ) ){
            return "VoiceOverlap";
        }else if( name.equals( "envelope" ) ){
            return "Envelope";
        }else if( name.equals( "moduration" ) ){
            return "Moduration";
        }else if( name.equals( "flags" ) ){
            return "Flags";
        }
        return name;
    }

    public Object clone() {
        UstEvent ret = new UstEvent();
        ret.length = length;
        ret.lyric = lyric;
        ret.note = note;
        ret.intensity = intensity;
        ret.pbType = pbType;
        if ( pitches != null ) {
            ret.pitches = new float[pitches.length];
            for ( int i = 0; i < pitches.length; i++ ) {
                ret.pitches[i] = pitches[i];
            }
        }
        ret.tempo = tempo;
        if ( vibrato != null ) {
            ret.vibrato = (UstVibrato)vibrato.clone();
        }
        if ( portamento != null ) {
            ret.portamento = (UstPortamento)portamento.clone();
        }
        if ( envelope != null ) {
            ret.envelope = (UstEnvelope)envelope.clone();
        }
        ret.preUtterance = preUtterance;
        ret.voiceOverlap = voiceOverlap;
        ret.flags = flags;
        ret.moduration = moduration;
        ret.tag = tag;
        return ret;
    }

    public void print( StreamWriter sw, int index ) throws IOException{
        sw.writeLine( (new DecimalFormat( "0000" )).format( index ) );
        sw.writeLine( "Length=" + length );
        sw.writeLine( "Lyric=" + lyric );
        sw.writeLine( "NoteNum=" + note );
        if ( intensity >= 0 ) {
            sw.writeLine( "Intensity=" + intensity );
        }
        if ( pbType >= 0 && pitches != null ) {
            sw.writeLine( "PBType=" + pbType );
            sw.write( "Piches=" );
            for ( int i = 0; i < pitches.length; i++ ) {
                if ( i == 0 ) {
                    sw.write( pitches[i] + "" );
                } else {
                    sw.write( "," + pitches[i] );
                }
            }
            sw.writeLine();
        }
        if ( tempo > 0 ) {
            sw.writeLine( "Tempo=" + tempo );
        }
        if ( vibrato != null ) {
            sw.writeLine( vibrato.toString() );
        }
        if ( portamento != null ) {
            portamento.print( sw );
        }
        if ( envelope != null ) {
            if ( preUtterance >= 0 ) {
                sw.writeLine( "PreUtterance=" + preUtterance );
            }
            if ( voiceOverlap != 0 ) {
                sw.writeLine( "VoiceOverlap=" + voiceOverlap );
            }
            sw.writeLine( envelope.toString() );
        }
        if ( !flags.equals( "" ) ) {
            sw.writeLine( "Flags=" + flags );
        }
        if ( moduration >= 0 ) {
            sw.writeLine( "Moduration=" + moduration );
        }
    }
}

File: Boare.Lib.Vsq/UstEvent.cs
/*
 * UstEvent.cs
 * Copyright (c) 2009 kbinani
 *
 * This file is part of Boare.Lib.Vsq.
 *
 * Boare.Lib.Vsq is free software; you can redistribute it and/or
 * modify it under the terms of the BSD License.
 *
 * Boare.Lib.Vsq is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
 */
#if JAVA
package org.kbinani.vsq;

import java.io.*;
import org.kbinani.*;
#else
using System;
using bocoree;
using bocoree.io;

namespace Boare.Lib.Vsq
{
#endif

#if JAVA
    public class UstEvent implements Cloneable, Serializable
#else
    [Serializable]
    public class UstEvent : ICloneable
#endif
    {
        public String Tag;
        public int Length = 0;
        public String Lyric = "";
        public int Note = -1;
        public int Intensity = -1;
        public int PBType = -1;
        public float[] Pitches = null;
        public float Tempo = -1;
        public UstVibrato Vibrato = null;
        public UstPortamento Portamento = null;
        public int PreUtterance = 0;
        public int VoiceOverlap = 0;
        public UstEnvelope Envelope = null;
        public String Flags = "";
        public int Moduration = 100;
        public int Index;

        public UstEvent()
        {
        }

        public int getLength()
        {
            return Length;
        }

        public void setLength( int value )
        {
            Length = value;
        }

        public Object clone()
        {
            UstEvent ret = new UstEvent();
            ret.setLength( Length );
            ret.Lyric = Lyric;
            ret.Note = Note;
            ret.Intensity = Intensity;
            ret.PBType = PBType;
            if ( Pitches != null )
            {
                ret.Pitches = new float[Pitches.Length];
                for ( int i = 0; i < Pitches.Length; i++ )
                {
                    ret.Pitches[i] = Pitches[i];
                }
            }
            ret.Tempo = Tempo;
            if ( Vibrato != null )
            {
                ret.Vibrato = (UstVibrato)Vibrato.clone();
            }
            if ( Portamento != null )
            {
                ret.Portamento = (UstPortamento)Portamento.clone();
            }
            if ( Envelope != null )
            {
                ret.Envelope = (UstEnvelope)Envelope.clone();
            }
            ret.PreUtterance = PreUtterance;
            ret.VoiceOverlap = VoiceOverlap;
            ret.Flags = Flags;
            ret.Moduration = Moduration;
            ret.Tag = Tag;
            return ret;
        }

#if !JAVA
        public object Clone()
        {
            return clone();
        }
#endif

        public void print( BufferedWriter sw )
#if JAVA
            throws IOException
#endif
        {
            if ( this.Index == int.MinValue )
            {
                sw.write( "[#PREV]" );
                sw.newLine();
            }
            else if ( this.Index == int.MaxValue )
            {
                sw.write( "[#NEXT]" );
                sw.newLine();
            }
            else
            {
                sw.write( "[#" + PortUtil.formatDecimal( "0000", Index ) );
                sw.newLine();
            }
            sw.write( "Length=" + Length );
            sw.newLine();
            sw.write( "Lyric=" + Lyric );
            sw.newLine();
            sw.write( "NoteNum=" + Note );
            sw.newLine();
            if ( Intensity >= 0 )
            {
                sw.write( "Intensity=" + Intensity );
                sw.newLine();
            }
            if ( PBType >= 0 && Pitches != null )
            {
                sw.write( "PBType=" + PBType );
                sw.newLine();
                sw.write( "Piches=" );
                for ( int i = 0; i < Pitches.Length; i++ )
                {
                    if ( i == 0 )
                    {
                        sw.write( Pitches[i] + "" );
                    }
                    else
                    {
                        sw.write( "," + Pitches[i] );
                    }
                }
                sw.newLine();
            }
            if ( Tempo > 0 )
            {
                sw.write( "Tempo=" + Tempo );
                sw.newLine();
            }
            if ( Vibrato != null )
            {
                sw.write( Vibrato.ToString() );
                sw.newLine();
            }
            if ( Portamento != null )
            {
                Portamento.print( sw );
            }
            if ( Envelope != null )
            {
                if ( PreUtterance >= 0 )
                {
                    sw.write( "PreUtterance=" + PreUtterance );
                    sw.newLine();
                }
                if ( VoiceOverlap != 0 )
                {
                    sw.write( "VoiceOverlap=" + VoiceOverlap );
                    sw.newLine();
                }
                sw.write( Envelope.ToString() );
                sw.newLine();
            }
            if ( Flags != "" )
            {
                sw.write( "Flags=" + Flags );
                sw.newLine();
            }
            if ( Moduration >= 0 )
            {
                sw.write( "Moduration=" + Moduration );
                sw.newLine();
            }
        }
    }

#if !JAVA
}
#endif
